File: packages/case-definition-dsl/src/matchers/convenience/ReferenceMatch.ts
import { LOOKUP_MATCHER_TYPE } from '@contract-case/case-plugin-base';
import { AnyMatcher } from '../base';

/**
 * Matches a named matcher created with `NamedMatch`.
 * @public
 */
export class ReferenceMatch extends AnyMatcher {
  /** @internal */
  readonly '_case:matcher:type': typeof LOOKUP_MATCHER_TYPE;

  /** @internal */
  readonly '_case:matcher:uniqueName': string;

  /**
   * @param name - The name you gave to a previous call of `NamedMatch`
   */
  constructor(name: string) {
    super(LOOKUP_MATCHER_TYPE);
    this['_case:matcher:uniqueName'] = name;
  }

  /**
   * For non-TypeScript implementations (see `AnyMatcher.toJSON`)
   *
   * @privateRemarks
   * This comment and the implementation is boilerplate on all matchers to avoid
   * outputting duplicate unimportant documentation on all matcher classes of
   * the docs. Only modify this comment or the implementation via search and replace.
   */
  override toJSON(): unknown {
    return super.toJSON();
  }
}

File: packages/dsl-java/src/main/java/io/contract_testing/contractcase/dsl/matchers/convenience/ReferenceMatch.java
package io.contract_testing.contractcase.dsl.matchers.convenience;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.contract_testing.contractcase.dsl.ContractCaseDsl;
import io.contract_testing.contractcase.dsl.DslMatcher;
import jakarta.annotation.Generated;
import java.lang.String;
import lombok.Builder;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;

/**
 * Matches a named matcher created with the NamedMatch matcher.
 */
@Generated("@contract-case/case-definition-generator")
@ContractCaseDsl
public class ReferenceMatch implements DslMatcher {

  /**
   * ContractCase's internal type for this element
   */
  @Getter
  @JsonProperty("_case:matcher:type")
  private final String type;

  /**
   * The name you gave to a previous call of NamedMatch
   */
  @Getter
  @JsonProperty("_case:matcher:uniqueName")
  private final String uniqueName;

  /**
   * Matches a named matcher created with the NamedMatch matcher.
   * @param uniqueName The name you gave to a previous call of NamedMatch
   */
  @Builder
  public ReferenceMatch(@NotNull final String uniqueName) {
    this.type = "_case:Lookup";
    this.uniqueName = uniqueName;
  }
}
